src/ConfigCatProvider.tsx

```diff
@@ -15,39 +17,33 @@ type ConfigCatProviderProps = {
 };
 
 type ConfigCatProviderState = {
-  client: IConfigCatClient;
+  client?: IConfigCatClient;
   lastUpdated?: Date;
 };
 
 const initializedClients = new Map<string, number>();
 
-class ConfigCatProvider extends Component<
-/* eslint-disable @typescript-eslint/indent */
-  PropsWithChildren<ConfigCatProviderProps>,
-  ConfigCatProviderState,
-  {}
-/* eslint-enable @typescript-eslint/indent */
-> {
+class ConfigCatProvider extends Component<PropsWithChildren<ConfigCatProviderProps>, ConfigCatProviderState, {}> {
 
   constructor(props: ConfigCatProviderProps) {
     super(props);
-    const client = this.initializeConfigCatClient();
-    this.state = { client };
+    this.state = {};
   }
 
   componentDidMount(): void {
-    this.state?.client?.on("clientReady", () => this.clientReady());
-    this.state?.client?.on("configChanged", newConfig => this.reactConfigChanged(newConfig));
+    this.setState({ client: this.initializeConfigCatClient(), lastUpdated: new Date() });
+    this.state.client?.on("clientReady", () => this.clientReady());
+    this.state.client?.on("configChanged", newConfig => this.reactConfigChanged(newConfig));
   }
 
   componentWillUnmount(): void {
-    this.state?.client?.removeListener("clientReady", () => this.clientReady());
-    this.state?.client?.removeListener("configChanged", newConfig => this.reactConfigChanged(newConfig));
+    this.state.client?.removeListener("clientReady", () => this.clientReady());
+    this.state.client?.removeListener("configChanged", newConfig => this.reactConfigChanged(newConfig));
```

Comment:
Also tangential: these calls have no effect as we want to remove *different instances* of handler functions from the ones we subscribed.

To make this work, we need to store the handlers into variables and pass those to `removeListener`.